docs/src/tutorials/arrays_colors.md

```diff
@@ -222,6 +222,57 @@ These values are close to the channels of `c`, but have been rounded
 off---each channel is encoded with only 8 bits, so some approximation
 of the exact floating-point value is unavoidable.
 
+## Preserving colorant type while conversion
+
+While using `Float32`, `Float64` to convert the raw storage type of an image, 
+the color space information is lost. Replacaing `Float32`, `Float64` with their lower-case
+forms preserves the colorant type.
+
+- Grayscale Images
+
+```jldoctest; setup = :(using TestImages, ImageCore; img = testimage("cameraman");)
+julia> summary(img)
+"512×512 Array{Gray{N0f8},2} with eltype Gray{Normed{UInt8,8}}"
+
+julia> summary(float32.(img))
+"512×512 Array{Gray{Float32},2} with eltype Gray{Float32}"
+
+julia> summary(Float32.(img))
+"512×512 Array{Float32,2}"
+```
+- Color Images
+
+```jldoctest; setup = :(using TestImages, ImageCore; img = testimage("fabio_color_256");)
+julia> summary(img)
+"256×256 Array{RGB{N0f8},2} with eltype RGB{Normed{UInt8,8}}"
+
+julia> summary(float32.(img))
+"256×256 Array{RGB{Float32},2} with eltype RGB{Float32}"
+
+julia> summary(Float32.(img))
+ERROR: MethodError: no method matching Float32(::RGB{Normed{UInt8,8}})
+Closest candidates are:
+  Float32(::Int8) at float.jl:60
+  Float32(::Int16) at float.jl:60
+  Float32(::Int32) at float.jl:60
+  ...
+  Stacktrace:
+  [...]
+```
+**Note**: `Float64.(rgb_img)` throws error because it does not implicitly   
+convert RGB to single channel but `float64.(rgb_img)` works as expected.
```

Comment:
"Works as expected" depends entirely on what your expectations are. Some people will expect it to return a single floating-point number.

How about copy/pasting the examples with a single colorant object that I gave in my explanation to you? This mixes together broadcasting with colorspace conversion, and we don't have to mix the concepts.